### ISSUE TITLE
UCP: support atomics of various sizes

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#684]
1. Emulate atomics over active messages
2. Use 2 atomics of 64-bit to emulate one atomic of smaller size.

**Comment by shamisp** [comment]
@yosefe - please coordinate with @MattBBaker since he has been looking into porting some of these features from UCCS.

**Comment by MattBBaker** [comment]
Yes, I've been looking into how to do this. I was thinking taking the UCCS code, make that UCS functions and filling in the operations tables in UCT with the emulation functions, with ugni_rdma over writing those with native functions if Aries is supported. As long as the flags are honest, and emulation functions are populated, then wireup should be able to score based on what is natively supported.
